### ISSUE TITLE
Add image options for paper wallets

### ISSUE THREAD

**Issue by jonstern** [pointbiz/bitaddress.org#97]
I would love the opportunity to make custom paper wallet backgrounds and choose them in the paper wallet tab. I'm not a coder, but please let me know if there is a way to do this now that a normal person can do and what tools to use. Thanks.

**Comment by mathiasrw** [comment]
Good point.

**Comment by chris-belcher** [comment]
It might be possible in the future to add a GUI interface to the web page that allows someone to use their own image.

Until then, it's possible to edit the html to add your own image. You said you're not a coder but I'll describe the process anyway.

1. Download bitaddress.org.html and open in a text editor

2. Press CTRL+F and search for 'btcprivwif'; including the ' marks and ; semicolon.

3. You'll see code that looks a bit like this, for me it's on line 9777

	templateArtisticHtml: function (i) {
			var keyelement = 'btcprivwif';
			var image;
			if (ninja.wallets.paperwallet.encrypt) {
				keyelement = 'btcencryptedkey'
				image = 'data:image/png;base64,LOTS OF BASE64 CHARACTERS HERE';
			}
			else {
				image = 'data:image/png;base64,LOTS OF BASE64 CHARACTERS HERE';
			}

That's an if-statement which controls whether the encrypted bip38 graphic appears or the nonencrypted graphic.

4. Create your image that has the same size as the paper wallet graphic, as well as space for the QR codes. Upload to https://www.base64-image.de/ (first result on google) to get a base64-encoded version of your image.

5. Replace the existing base64-encoded image (one of the parts where it says LOTS OF BASE64 CHARACTERS HERE) with your own base64-encoded image. 

6. Save and open in a web browser, the image should appear if you didn't mess anything up.

Be warned, the LOTS OF BASE64 CHARACTERS HERE part has a huge amount of characters. Like 177kB worth.

Here's my version with very basic graphics skills.

![epaperwallet-checkem-scrooge](https://cloud.githubusercontent.com/assets/8398185/10556519/1e08e7c8-747d-11e5-9bfd-a9950964b05b.png)

**Comment by 100000s** [comment]
> 2\. 'btcprivwif';

I'm not sure what I'm doing wrong. Should the base64 image be used to replace both sets of images in the code? Should it (or they) end in a quotation mark and semicolon? Is the image not working because it is a couple of pixels narrower?

**Comment by chris-belcher** [comment]
Wow this issue is old.

Don't use paper wallets, see the arguments for why here: https://en.bitcoin.it/wiki/Paper_wallet

If you want to store bitcoins on paper you're much better off using a proper wallet that creates a seed phrase, and then write that down.

**Comment by pointbiz** [comment]
Paper wallets are still a valid way to store your Bitcoin. Yes, there are pros and cons but that article is full of bias and inaccuracies. 

I think mnemonic seeds are great but have drawbacks too.

Keys have advantages over seed phrases. They are more universally supported and you are more likely to be able to spend from a key than a seed phrase 10 or 20 years from now.

Seed phrases have no universal standard (there's a few incomplete standards that are incorrectly implemented in the wild). Keys have well determined standards. The seed encoding doesn't include important information like the derivation path. Therefore, a seed from one wallet is not guaranteed to work in another wallet. A key however will work in all wallets that support sweeping keys.

**Comment by 100000s** [comment]
I'd like to improve on secure paper wallets. If this design were combined with the security measures of bitaddress (generating wallets offline), they are still, I believe, usable. 

![image](https://user-images.githubusercontent.com/86668971/124637251-7f0cd300-de57-11eb-83b8-d1cad51f5db1.png)

Caveats aside, is it still possible to change the graphic as discussed above?

**Comment by pointbiz** [comment]
Yes. The instructions above from chris are still valid. You need to have a single quote and semi-colon at the end of the base64 data. I believe it should still work if the image is a few pixels off.

You can use the same image data in both spots. There's two images in the original. Yellow for unencrypted addresses and blue for the BIP38 encrypted addresses. 

The image you are trying to use above is very different in size and you'll have to update some CSS to position the address and keys where you want them for that design.

**Comment by 100000s** [comment]
> Yes. The instructions above from chris are still valid. You need to have a single quote and semi-colon at the end of the base64 data. I believe it should still work if the image is a few pixels off.
> 
> You can use the same image data in both spots. There's two images in the original. Yellow for unencrypted addresses and blue for the BIP38 encrypted addresses.
> 
> The image you are trying to use above is very different in size and you'll have to update some CSS to position the address and keys where you want them for that design.

Thanks pointbiz! 

I've tried that with the following design, but, for some reason it removes the QR code and keeps the original design. Maybe I'm using the wrong browser or encoder ( https://www.base64encode.net/base64-image-encoder )? 

Here is my image. It's just a few pixels off: 

![btccheck1 1sp261](https://user-images.githubusercontent.com/86668971/124641423-b16cff00-de5c-11eb-8505-0ea729db048f.png)

**Comment by 100000s** [comment]
> Paper wallets are still a valid way to store your Bitcoin. Yes, there are pros and cons but **that article is full of bias** and inaccuracies.

[You ain't kidding, pointbiz!](https://en.bitcoin.it/w/index.php?title=Paper_wallet&action=history)

**Comment by BitAddr-org** [comment]
⚠️ This repository is no longer actively maintained.

Please open new issues and submit contributions on the maintained fork:

- Repository: [TheophileWalter/bitaddr.org](https://github.com/TheophileWalter/bitaddr.org/)
- Website: [bitaddr.org](https://bitaddr.org/)

Thank you.
